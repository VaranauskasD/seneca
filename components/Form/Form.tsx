import React from 'react'
import styled from 'styled-components'

import { RadioGroup } from '../RadioGroup'

interface FormProps {
  question: string
  optionGroups: { option: string; correct?: boolean }[][]
  handleOption: (optionGroupId: number, correct: boolean) => void
  isAnswered: boolean
  dynamicRatio: number
}

const StyledQuestion = styled.h1`
  color: ${(props) => `${props.theme.colors.primary}`};
`

const StyledAnswer = styled.h2`
  color: ${(props) => `${props.theme.colors.primary}`};
`

const StyledForm = styled.form`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  text-align: center;
`

const RadioWrapper = styled.div``

export const Form = (props: FormProps) => {
  return (
    <React.Fragment>
      <StyledForm>
        <StyledQuestion>{props.question}</StyledQuestion>
        {props.optionGroups.map((optionGroup, key) => (
          <RadioWrapper key={`optionGroup-${key}`}>
            <RadioGroup
              id={key}
              options={optionGroup}
              handleOption={props.handleOption}
              dynamicRatio={props.dynamicRatio}
              isAnswered={props.isAnswered}
            />
          </RadioWrapper>
        ))}
      </StyledForm>
      <StyledAnswer>
        The answer is {props.isAnswered ? `correct` : `incorrect`}
      </StyledAnswer>
    </React.Fragment>
  )
}

export default Form
