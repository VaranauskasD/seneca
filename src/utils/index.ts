export { theme } from './theme'
export { customRender } from './test'
