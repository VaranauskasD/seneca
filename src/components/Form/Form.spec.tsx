import { screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import { customRender as render } from '../../utils/test'
import { Form, FormProps } from './Form'

let mockFormProps: FormProps
beforeEach(() => {
  mockFormProps = {
    question: 'An animal cell contains',
    optionGroups: [
      [{ option: 'Cell wall', correct: true }, { option: 'Ribosomes' }],
      [{ option: 'Cytoplasm', correct: true }, { option: 'Chloroplast' }],
      [
        { option: 'Partially permeable membrane', correct: true },
        { option: 'Impermeable membrane' },
      ],
      [{ option: 'Cellulose' }, { option: 'Mitochondria', correct: true }],
      [
        { option: 'Cellulose' },
        { option: 'Ribosomes' },
        { option: 'Mitochondria', correct: true },
        { option: 'Chroloplast' },
      ],
    ],
    handleOption: () => {},
    isAnswered: false,
    dynamicRatio: 0,
  }
})

describe('Form', () => {
  it('renders correctly', async () => {
    const { asFragment } = render(<Form {...mockFormProps} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders correctly when answer is reached', async () => {
    mockFormProps.isAnswered = true
    const { asFragment } = render(<Form {...mockFormProps} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
