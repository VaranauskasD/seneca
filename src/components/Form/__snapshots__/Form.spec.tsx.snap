// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form renders correctly 1`] = `
<DocumentFragment>
  <form
    class="sc-gKXOVf eUHymF"
  >
    <h1
      class="sc-eCYdqJ fAZKGE"
    >
      An animal cell contains
    </h1>
    <div
      class="sc-iBkjds"
    >
      <div
        class="sc-hKMtZM fRXuCA"
      >
        <div
          class="sc-bczRLJ fNFKHi"
        >
          <label
            class="sc-dkzDqf hdgGtK"
          >
            <input
              checked=""
              class="sc-gsnTZi exMZOs"
              id="option-0-0"
              name="option-0"
              type="radio"
            />
            Cell wall
          </label>
        </div>
        <div
          class="sc-bczRLJ jNqizO"
        >
          <label
            class="sc-dkzDqf hdgGtK"
          >
            <input
              class="sc-gsnTZi exMZOs"
              id="option-0-1"
              name="option-0"
              type="radio"
            />
            Ribosomes
          </label>
        </div>
      </div>
    </div>
    <div
      class="sc-iBkjds"
    >
      <div
        class="sc-hKMtZM fRXuCA"
      >
        <div
          class="sc-bczRLJ fNFKHi"
        >
          <label
            class="sc-dkzDqf hdgGtK"
          >
            <input
              checked=""
              class="sc-gsnTZi exMZOs"
              id="option-1-0"
              name="option-1"
              type="radio"
            />
            Cytoplasm
          </label>
        </div>
        <div
          class="sc-bczRLJ jNqizO"
        >
          <label
            class="sc-dkzDqf hdgGtK"
          >
            <input
              class="sc-gsnTZi exMZOs"
              id="option-1-1"
              name="option-1"
              type="radio"
            />
            Chloroplast
          </label>
        </div>
      </div>
    </div>
    <div
      class="sc-iBkjds"
    >
      <div
        class="sc-hKMtZM fRXuCA"
      >
        <div
          class="sc-bczRLJ fNFKHi"
        >
          <label
            class="sc-dkzDqf hdgGtK"
          >
            <input
              checked=""
              class="sc-gsnTZi exMZOs"
              id="option-2-0"
              name="option-2"
              type="radio"
            />
            Partially permeable membrane
          </label>
        </div>
        <div
          class="sc-bczRLJ jNqizO"
        >
          <label
            class="sc-dkzDqf hdgGtK"
          >
            <input
              class="sc-gsnTZi exMZOs"
              id="option-2-1"
              name="option-2"
              type="radio"
            />
            Impermeable membrane
          </label>
        </div>
      </div>
    </div>
    <div
      class="sc-iBkjds"
    >
      <div
        class="sc-hKMtZM fRXuCA"
      >
        <div
          class="sc-bczRLJ fNFKHi"
        >
          <label
            class="sc-dkzDqf hdgGtK"
          >
            <input
              checked=""
              class="sc-gsnTZi exMZOs"
              id="option-3-0"
              name="option-3"
              type="radio"
            />
            Cellulose
          </label>
        </div>
        <div
          class="sc-bczRLJ jNqizO"
        >
          <label
            class="sc-dkzDqf hdgGtK"
          >
            <input
              class="sc-gsnTZi exMZOs"
              id="option-3-1"
              name="option-3"
              type="radio"
            />
            Mitochondria
          </label>
        </div>
      </div>
    </div>
    <div
      class="sc-iBkjds"
    >
      <div
        class="sc-hKMtZM efHAkz"
      >
        <div
          class="sc-bczRLJ bDYwXX"
        >
          <label
            class="sc-dkzDqf hdgGtK"
          >
            <input
              checked=""
              class="sc-gsnTZi exMZOs"
              id="option-4-0"
              name="option-4"
              type="radio"
            />
            Cellulose
          </label>
        </div>
        <div
          class="sc-bczRLJ hXUtLi"
        >
          <label
            class="sc-dkzDqf hdgGtK"
          >
            <input
              class="sc-gsnTZi exMZOs"
              id="option-4-1"
              name="option-4"
              type="radio"
            />
            Ribosomes
          </label>
        </div>
        <div
          class="sc-bczRLJ hXUtLi"
        >
          <label
            class="sc-dkzDqf hdgGtK"
          >
            <input
              class="sc-gsnTZi exMZOs"
              id="option-4-2"
              name="option-4"
              type="radio"
            />
            Mitochondria
          </label>
        </div>
        <div
          class="sc-bczRLJ iGNBNP"
        >
          <label
            class="sc-dkzDqf hdgGtK"
          >
            <input
              class="sc-gsnTZi exMZOs"
              id="option-4-3"
              name="option-4"
              type="radio"
            />
            Chroloplast
          </label>
        </div>
      </div>
    </div>
  </form>
  <h2
    class="sc-jSMfEi kTCAsy"
  >
    The answer is incorrect
  </h2>
</DocumentFragment>
`;

exports[`Form renders correctly when answer is reached 1`] = `
<DocumentFragment>
  <form
    class="sc-gKXOVf eUHymF"
  >
    <h1
      class="sc-eCYdqJ fAZKGE"
    >
      An animal cell contains
    </h1>
    <div
      class="sc-iBkjds"
    >
      <div
        class="sc-hKMtZM fRXuCA"
      >
        <div
          class="sc-bczRLJ fNFKHi"
        >
          <label
            class="sc-dkzDqf eThtLg"
          >
            <input
              checked=""
              class="sc-gsnTZi exMZOs"
              disabled=""
              id="option-0-0"
              name="option-0"
              type="radio"
            />
            Cell wall
          </label>
        </div>
        <div
          class="sc-bczRLJ jNqizO"
        >
          <label
            class="sc-dkzDqf eThtLg"
          >
            <input
              class="sc-gsnTZi exMZOs"
              disabled=""
              id="option-0-1"
              name="option-0"
              type="radio"
            />
            Ribosomes
          </label>
        </div>
      </div>
    </div>
    <div
      class="sc-iBkjds"
    >
      <div
        class="sc-hKMtZM fRXuCA"
      >
        <div
          class="sc-bczRLJ fNFKHi"
        >
          <label
            class="sc-dkzDqf eThtLg"
          >
            <input
              checked=""
              class="sc-gsnTZi exMZOs"
              disabled=""
              id="option-1-0"
              name="option-1"
              type="radio"
            />
            Cytoplasm
          </label>
        </div>
        <div
          class="sc-bczRLJ jNqizO"
        >
          <label
            class="sc-dkzDqf eThtLg"
          >
            <input
              class="sc-gsnTZi exMZOs"
              disabled=""
              id="option-1-1"
              name="option-1"
              type="radio"
            />
            Chloroplast
          </label>
        </div>
      </div>
    </div>
    <div
      class="sc-iBkjds"
    >
      <div
        class="sc-hKMtZM fRXuCA"
      >
        <div
          class="sc-bczRLJ fNFKHi"
        >
          <label
            class="sc-dkzDqf eThtLg"
          >
            <input
              checked=""
              class="sc-gsnTZi exMZOs"
              disabled=""
              id="option-2-0"
              name="option-2"
              type="radio"
            />
            Partially permeable membrane
          </label>
        </div>
        <div
          class="sc-bczRLJ jNqizO"
        >
          <label
            class="sc-dkzDqf eThtLg"
          >
            <input
              class="sc-gsnTZi exMZOs"
              disabled=""
              id="option-2-1"
              name="option-2"
              type="radio"
            />
            Impermeable membrane
          </label>
        </div>
      </div>
    </div>
    <div
      class="sc-iBkjds"
    >
      <div
        class="sc-hKMtZM fRXuCA"
      >
        <div
          class="sc-bczRLJ fNFKHi"
        >
          <label
            class="sc-dkzDqf eThtLg"
          >
            <input
              checked=""
              class="sc-gsnTZi exMZOs"
              disabled=""
              id="option-3-0"
              name="option-3"
              type="radio"
            />
            Cellulose
          </label>
        </div>
        <div
          class="sc-bczRLJ jNqizO"
        >
          <label
            class="sc-dkzDqf eThtLg"
          >
            <input
              class="sc-gsnTZi exMZOs"
              disabled=""
              id="option-3-1"
              name="option-3"
              type="radio"
            />
            Mitochondria
          </label>
        </div>
      </div>
    </div>
    <div
      class="sc-iBkjds"
    >
      <div
        class="sc-hKMtZM efHAkz"
      >
        <div
          class="sc-bczRLJ bDYwXX"
        >
          <label
            class="sc-dkzDqf eThtLg"
          >
            <input
              checked=""
              class="sc-gsnTZi exMZOs"
              disabled=""
              id="option-4-0"
              name="option-4"
              type="radio"
            />
            Cellulose
          </label>
        </div>
        <div
          class="sc-bczRLJ hXUtLi"
        >
          <label
            class="sc-dkzDqf eThtLg"
          >
            <input
              class="sc-gsnTZi exMZOs"
              disabled=""
              id="option-4-1"
              name="option-4"
              type="radio"
            />
            Ribosomes
          </label>
        </div>
        <div
          class="sc-bczRLJ hXUtLi"
        >
          <label
            class="sc-dkzDqf eThtLg"
          >
            <input
              class="sc-gsnTZi exMZOs"
              disabled=""
              id="option-4-2"
              name="option-4"
              type="radio"
            />
            Mitochondria
          </label>
        </div>
        <div
          class="sc-bczRLJ iGNBNP"
        >
          <label
            class="sc-dkzDqf eThtLg"
          >
            <input
              class="sc-gsnTZi exMZOs"
              disabled=""
              id="option-4-3"
              name="option-4"
              type="radio"
            />
            Chroloplast
          </label>
        </div>
      </div>
    </div>
  </form>
  <h2
    class="sc-jSMfEi kTCAsy"
  >
    The answer is correct
  </h2>
</DocumentFragment>
`;
